rag/app/book.py

```
@@ -157,13 +175,20 @@ def chunk(filename, binary=None, from_page=0, to_page=MAXIMUM_PAGE_NUMBER, lang=
     bull = bullets_category([t for t in random_choices([t for t, _ in sections], k=100)])
-    if bull >= 0:
+    if bull >= 0 and not parser_config.get("chunk_token_num"):
+        logging.debug("book chunk policy: hierarchical_merge (chunk_token_num unset)")
         chunks = ["\n".join(ck) for ck in hierarchical_merge(bull, sections, 5)]
     else:
-        sections = [s.split("@") for s, _ in sections]
-        sections = [(pr[0], "@" + pr[1]) if len(pr) == 2 else (pr[0], "") for pr in sections]
-        chunks = naive_merge(sections, parser_config.get("chunk_token_num", 256), parser_config.get("delimiter", "\n。；！？"))
+        chunk_token_num = parser_config.get("chunk_token_num", 256)
+        logging.debug("book chunk policy: naive_merge (chunk_token_num=%s)", chunk_token_num)
+        chunks = naive_merge(_sections_with_positions(sections), chunk_token_num, parser_config.get("delimiter", "\n。；！？"))
```

Ruff (0.15.21)

[warning] String contains ambiguous `；` (FULLWIDTH SEMICOLON). Did you mean `;` (SEMICOLON)?

(RUF001)